tests/test1165.pl

```diff
@@ -44,7 +44,7 @@ sub scanconf {
     my ($f)=@_;
     open S, "<$f";
     while(<S>) {
-        if(/(CURL_DISABLE_[A-Z_]+)/g) {
+        if(/(?:^|[^A-Za-z0-9\-])(CURL_DISABLE_[A-Z0-9_]+)(?:$|[^A-Za-z0-9\-])/g) {
```

Comment:
This cannot be correct, since it puts the wrong part into `$1`.

Why not just extend the pattern? Surely it doesn't do any false positive matches elsewhere?
~~~perl
 if(/(CURL_DISABLE_[A-Z0-9_]+)/g) {

Reply:
Nope. :)
It is non-capturing group because of `?:`
`(?:^|[^A-Za-z0-9\-])` is *matched* but not *captured*.
The same for `(?:$|[^A-Za-z0-9\-])`.

It works correctly, I tested it with modified script version with printing all matched results.
Right now there are no false positives here even without the start and end patterns. But this is the test, it **must** catch incorrect code changes.

Reply:
Ok thanks, that goes beyond what I knew. I still think it's overly complicated for no good reason.